learning-another-language/variables.md

```diff
@@ -68,6 +68,41 @@ var jingsFavColor = 'green';
 ```
 <!-- prettier-ignore-end -->
 
+### Scope of var vs let & const
+
+When creating a variable using `let` and `const` will scope the variable to the block it is in.  In the below example `i` and `name` are defined inside of the `for-of` loop and are scoped to the loop so are not accessable outside of the loop.  This applies to all blocks of code in JavaScript, except conditionals (if statements).  
+
+```javaScript
+const nums = [14, 15, 16, 18, 19, 20];
+
+for (let i of nums) {
+    console.log(`i is ${i}`);
+    const name = 'peter';
+    console.log(`name is ${name}`);
+}
+
+console.log(i); // ReferenceError: i is not defined
+console.log(name); // ReferenceError: name is not defined
+
+```
+
+On the other hand, when using `var` the variable is scoped to the function it is defined in.
+
+```javascript
+const nums = [14, 15, 16, 18, 19, 20];
+
+for (var i of nums) {
+    console.log(`i is ${i}`);
+    var name = 'peter';
+    console.log(`name is ${name}`);
+}
+
+console.log(i); // 20
+console.log(name); // peter
+```
+
+This can be very confusing!  Variables defined with `let` and `const` are scoped to the block they are defined in, whereas variables created with `var` are scoped to the function.  This can be confusing and can lead to bugs.  In general, i.e. **always**, it is best to use `let` and `const`.
```

Comment:
Reworded to avoid the redundant-sounding double usage of confusing. Interjected i.e. is a punctuation mess (should be followed by a comma), so reworded to avoid that.

And single spaces after period.

```suggestion
This can be very confusing! Variables defined with `let` and `const` are scoped to the block they are defined in, whereas variables created with `var` are scoped to the function, or to the global scope. This is easy to overlook and can lead to bugs. As a result, it is a best practice to **always** use `let` and `const`.
```